src/v2/v2_types.rs

```diff
@@ -0,0 +1,185 @@
+                    Ok((extract_node_info_with_node_type(status.node_info)?, ConsensusStatus::ConsensusPassive))
+                },
+                // The consensus protocol is not running on the node.
+                // This only occurs when the node does not support the protocol on the chain or the node is a 'Bootstrapper'.
+                Some(super::generated::node_status::ConsensusStatus::ConsensusNotRunningStatus(status)) => Ok((extract_node_info_with_node_type(status.node_info)?, ConsensusStatus::ConsensusNotRunning)),
+                // The node has been configured with baker keys.
+                Some(super::generated::node_status::ConsensusStatus::BakerConsensusStatus(status)) => {
+                    let baking_status = match status.committee_status {
+                        // The node has been configured with baker keys however it is not currently baking and possilby never will.
+                        0 => Ok(BakingStatus::NotInCommittee),
+                        // The baker keys are registered however the baker is not in the committee 
+                        // for the current 'Epoch'.
+                        1 => Ok(BakingStatus::AddedButNotActiveInCommittee),
+                        // The node has been configured with baker keys that does not match the account.
+                        2 => Ok(BakingStatus::AddedButWrongKeys),
+                        // The node is member of the baking committee.
+                        3 => {
+                            let account_index = crate::types::AccountIndex::from(status.baker_id);
+                            let baker_id = crate::types::BakerId::from(account_index);                        
+                            if !status.is_finalizer {
+                                Ok(BakingStatus::Baker(baker_id))
+                            } else {
+                                Ok(BakingStatus::Finalizer(baker_id))
+                            }
+                        },
+                        _ => Err(NodeStatusError::MalformedCommitteeStatus),
+                    };                    
+                    Ok((extract_node_info_with_node_type(status.node_info)?, ConsensusStatus::ConsensusBaking(baking_status?)))
+                },
+                _ => Err(NodeStatusError::MissingConsensusStatus),
+            }?;
+
+        Ok(NodeStatus{
+            network_status,
+            node_info,
+            consensus_status,
+        })
+    }
+}
+
+fn extract_node_info(node_info: Option<super::generated::NodeInfo>, node_type: NodeType) -> Result<NodeInfo, NodeStatusError> {
+    if let Some(node_info) = node_info {    
+        // Parse the local time of the node. 
+        // The local time is provided as a UNIX timestamp with millisecond precision.    
+        let local_time = chrono::Utc.timestamp_millis(node_info.local_time.ok_or(NodeStatusError::MissingLocaltime)?.timestamp as i64); // todo: proper error handling
```

Comment:
In the gRPC you defined the timestamp to seconds from unix epoch.
If it is milliseconds we already have a type `generated::Timestamp` with a conversion into a chrono type.